ember-headless-form/src/components/-private/types.ts

```diff
@@ -55,7 +55,7 @@ export type FieldValidateCallback<
  * Internal structure to track used fields
  * @private
  */
-export interface FieldData<
+export interface FieldRegistrationData<
```

Comment:
I was splitting this up into two concerns, that's why I renamed this.

So there is information passed form the field component to the parent form component via the `registerField` action. Basically the field register itself with its parent form, so the form has data available that the dev passed to the field. Currently this is just the _field_ level validation callback, which the form needs to have available. This is what the `FieldRegistrationData` contains.

Then in this PR there is a new [`FieldData` class being added](https://github.com/CrowdStrike/ember-headless-form/pull/25/files#diff-71dca24e838c62279dc22478137e692716eddbccaf782f3901bfda818c5e67daR54-R71), which gets created when registering the field, so it uses `FieldRegistrationData`, but it also contains more than that (the `validationEnabled` tracked property), which the field does not need to care about.

So that's why we have these two different but related concepts now.

As mentioned earlier, I added a bunch of JSDoc comments, also here, so hopefully that becomes more clear now!

Reply:
Thanks for this!